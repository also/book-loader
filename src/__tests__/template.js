test('does simple interpolation', () => {
  const template = require('../template');
  const env = template.create();
  expect(env).toMatchSnapshot();
  const preprocessed = template.preprocess(env, '<%= variable1 %> and <%- variable2 %>');
  expect(env).toMatchSnapshot();
  expect(preprocessed).toMatchSnapshot();
  const postprocessed = template.postprocess(env, preprocessed);
  expect(postprocessed).toMatchSnapshot();
});

test('preprocess ignores non-strings', () => {
  const template = require('../template');
  const env = template.create();
  const preprocessed = template.preprocess(env, 7);
  expect(env).toMatchSnapshot();
  expect(preprocessed).toMatchSnapshot();
});

test('applies templates', () => {
  const template = require('../template');
  const result = template.apply('<%= variable1 %><%- variable2 %> and <%= variable3 %>');
  expect(result).toMatchSnapshot();
});

test('creates generators', () => {
  const template = require('../template');
  const generator = template.generator('<%= variable1 %><%- variable2 %> and <%= variable3 %>');
  for (const element of generator) {
    expect(element).toMatchSnapshot();
  }
});
