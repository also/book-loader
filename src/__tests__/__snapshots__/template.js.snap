// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies templates 1`] = `
"( variable1 ) +
    (escapeHtml( variable2 )) +
    \\" and \\" +
    ( variable3 )"
`;

exports[`creates generators 1`] = `
Object {
  "js": " variable1 ",
  "rawJs": " variable1 ",
}
`;

exports[`creates generators 2`] = `
Object {
  "js": "escapeHtml( variable2 )",
  "rawJs": " variable2 ",
}
`;

exports[`creates generators 3`] = `
Object {
  "raw": " and ",
}
`;

exports[`creates generators 4`] = `
Object {
  "js": " variable3 ",
  "rawJs": " variable3 ",
}
`;

exports[`does simple interpolation 1`] = `
Object {
  "jsTemplateReplacements": Array [],
}
`;

exports[`does simple interpolation 2`] = `
Object {
  "jsTemplateReplacements": Array [
    " variable1 ",
    "escapeHtml( variable2 )",
  ],
}
`;

exports[`does simple interpolation 3`] = `"~~ replacement 0 ~~ and ~~ replacement 1 ~~"`;

exports[`does simple interpolation 4`] = `
"\\"\\" +
    ( variable1 ) +
    \\" and \\" +
    (escapeHtml( variable2 )) +
    \\"\\""
`;

exports[`preprocess ignores non-strings 1`] = `
Object {
  "jsTemplateReplacements": Array [],
}
`;

exports[`preprocess ignores non-strings 2`] = `7`;
